src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableDataSource.cs

```diff
@@ -67,5 +81,55 @@ public void Dispose()
         {
             tableManager.RemoveSource(this);
         }
+
+        public void Add(IAnalysisIssueVisualization hotspot)
+        {
+            var entry = new HotspotTableEntry(hotspot);
+            tableEntries.Add(entry);
+
+            lock (sinks)
+            {
+                foreach (var sink in sinks)
+                {
+                    sink.AddEntries(new[] {entry});
+                }
+            }
+
+            var hotspotFilePaths = hotspot.GetAllLocations().Select(x => x.CurrentFilePath);
+            IssuesChanged?.Invoke(this, new IssuesChangedEventArgs(hotspotFilePaths));
+        }
+
+        public event EventHandler<IssuesChangedEventArgs> IssuesChanged;
+
+        public IEnumerable<IAnalysisIssueLocationVisualization> GetLocations(string filePath)
+        {
+            var matchingLocations = tableEntries
+                .Select(entry => entry.Identity as IAnalysisIssueVisualization)
+                .SelectMany(hotspotViz => hotspotViz.GetAllLocations())
+                .Where(locationViz => PathHelper.IsMatchingPath(locationViz.CurrentFilePath, filePath));
+
+            return matchingLocations;
+        }
+
+        public void Refresh(IEnumerable<string> affectedFilePaths)
+        {
+            var changedEntries = tableEntries
+                .Where(entry => affectedFilePaths.Any(p =>
+                    PathHelper.IsMatchingPath(p, (entry.Identity as IAnalysisIssueVisualization).CurrentFilePath)))
+                .ToList();
+
+            if (!changedEntries.Any())
```

Comment:
Minor: you've turned the result into a List so you can use the 'Count' property rather than the Any extension method (I thought there was a analysis rule for this?)

Reply:
I can't find any rule for this 🤔

Reply:
Ah. I have a third-party analyzer VSIX installed locally. It might be from that.